sections/dashboard/Overview/Overview.tsx

```diff
@@ -39,9 +48,116 @@ const Overview: FC = () => {
 	);
 	const [activeMarketsTab, setActiveMarketsTab] = useState<MarketsTab>(MarketsTab.FUTURES);
 
+	const { network, walletAddress, synthsMap } = Connector.useContainer();
+
+	const { tokenBalances } = useAppSelector(({ balances }) => balances);
+	// Only available on Optimism mainnet
+	const oneInchEnabled = network.id === 10;
+
+	const isL2 = useIsL2();
+
+	const [exchangeTokens, setExchangeTokens] = useState<any>([]);
+	const [kwentaBalance, setKwentaBalance] = useState(zeroBN);
+
+	useContractRead({
+		address: KWENTA_TOKEN_ADDRESS['10'],
+		abi: erc20ABI,
+		functionName: 'balanceOf',
+		args: [walletAddress!],
+		watch: false,
+		enabled: !!walletAddress && isL2,
+		onSettled(data, error) {
+			if (error) logError(error);
+			if (data) {
+				setKwentaBalance(weiFromWei(data));
+			}
+		},
+	});
+
+	const noKwentaFound = !Object.values(tokenBalances).find(
+		(token: any) => token.token.address.toLowerCase() === KWENTA_TOKEN_ADDRESS['10'].toLowerCase()
+	);
+
+	useEffect(() => {
+		(async () => {
+			const kwentaTokenObj = {
+				Kwenta: {
+					balance: kwentaBalance,
+					token: {
+						address: KWENTA_TOKEN_ADDRESS['10'].toLowerCase(),
+						symbol: 'KWENTA',
+						name: 'Kwenta',
+					},
+				},
+			};
+
+			const allTokens =
+				oneInchEnabled && noKwentaFound && kwentaBalance.gt(0)
+					? { ...kwentaTokenObj, ...tokenBalances }
+					: tokenBalances;
+
+			const exchangeBalances: {
+				name: string;
+				currencyKey: string;
+				balance: Wei;
+				address: string;
+			}[] = oneInchEnabled
+				? Object.values(allTokens)
+						.filter((token: any) => !synthsMap[token.token.symbol])
+						.map((value: any) => {
+							return {
+								name: value.token.name,
+								currencyKey: value.token.symbol,
+								balance: toWei(value.balance),
+								address:
+									value.token.address === ETH_ADDRESS ? ETH_COINGECKO_ADDRESS : value.token.address,
+							};
+						})
+				: [];
+
+			const coinGeckoPrices = await sdk.exchange.batchGetCoingeckoPrices(
+				oneInchEnabled
+					? [
+							...Object.values(tokenBalances).map((value: any) =>
+								value.token.address.toLowerCase()
+							),
+							noKwentaFound ? [KWENTA_TOKEN_ADDRESS['10'].toLowerCase()] : [],
+					  ]
+					: [],
+				true
+			);
+
+			const _exchangeTokens = exchangeBalances.map((exchangeToken) => {
+				const { name, currencyKey, balance, address } = exchangeToken;
+
+				const price = coinGeckoPrices ? toWei(coinGeckoPrices[address]?.usd?.toString()) : zeroBN;
+				const priceChange = coinGeckoPrices
+					? toWei(coinGeckoPrices[address]?.usd_24h_change?.toString()).div(100)
+					: zeroBN;
+
+				const usdBalance = balance.mul(price);
+
+				return {
+					synth: currencyKey,
+					description: name,
+					balance,
+					usdBalance,
+					price,
+					priceChange,
+				};
+			});
+
+			setExchangeTokens(_exchangeTokens);
+		})();
```

Comment:
Let's name the function instead of using an anon function and we should wrap this all in a try catch in case we get a failure, will cause the app to crash otherwise. Eventually we'll want to move this into redux but we can tackle that in a separate PR.